osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -418,10 +418,13 @@ private CaughtObject getDroppedObject(CaughtObject caughtObject)
 
         private void clearPlate(DroppedObjectAnimation animation)
         {
-            var droppedObjects = caughtObjectContainer.Children.Select(getDroppedObject).ToArray();
+            var caughtObjects = caughtObjectContainer.Children.ToArray();
 
             caughtObjectContainer.Clear(false);
 
+            // Use the already returned PoolableDrawables for new objects
+            var droppedObjects = caughtObjects.Select(getDroppedObject).ToArray();
```

Comment:
This feels sorta dangerous. I suppose the intent here is to first return the caught drawable objects to pool, so that they can be reused for the drop. But there is no strong guarantee that something else hasn't applied a new hitobject to the caught DHOs in the meantime after the DHOs have been returned to pool. So you could potentially end up dropping an object that was never dropped.

This seems salvageable if you dump `caughtObjectContainer.Children.Select(dho => dho.HitObject).ToArray()` instead rather than the DHOs raw. That eliminates any risk of a mix-up.

Reply:
I was thinking there would be some issue of this sort with this approach, still I'm not sure that situation could happen. Will adjust anyway just in case.

Reply:
Not actually sure how to go about this, as I would need to save stuff from `IHasCatchObjectState` and also the objects last position, which would look messy. Correct me if I'm wrong, but the scenario that you are proposing would only potentially happen if multiple objects were caught at the same time, which won't happen?

Reply:
> the scenario that you are proposing would only potentially happen if multiple objects were caught at the same time

I'm not sure where this assertion is coming from. It doesn't sound correct. Objects scrolling onto the screen will also retrieve objects from the pool.

Is this change even doing much work? I would wager that most of the gains here actually come from the lifetime set.

Reply:
> Objects scrolling onto the screen will also retrieve drawable objects from the pool.

Those are not part of the `Catcher` class. Here, only the objects placed on the catcher are being allocated.

> Is this change even doing much work? I would wager that most of the gains here actually come from the lifetime set.

When comparing the videos it seems to cut the allocated memory and draw calls in half.

Reply:
I think this is safe enough, based on the above assumption (the pools are local and usages are all in the method used here).